### DISCUSSION TITLE
Creating custom transitions between cross sections

### DISCUSSION THREAD

**Discussion by smgitrep** [gdsfactory/gdsfactory#3207] [Q&A]
I found the following code to create transitional cross sections between waveguides:

Xtrans = gf.path.transition(cross_section1=X1, cross_section2=X2, width_type="sine")

Is it possible to use my own custom transition?  For example, I have a set of widths vs propagation distance that defines the transition.  How can I pass the (width, propagation distance) data to gf.path.transition()?  If I can't pass the (width, propagation distance) data, can I pass a function instead?  For example, my_transition_width(propagation_distance)?

**Comment by joamatab** [comment]
why not just draw the polygons, if your transition is complex it may be easier to just define its polygons

**Reply by smgitrep** [reply]
For simple straight transitions, drawing a polygon is a simple solution.  I will give it a try.  Thank you.

If the transition path is not straight, is there another way to use custom width vs distance functions in GDSFACTORY?

**Reply by sebastian-goeldi** [reply]
I think gdsfactory also has a function (I don't know it like I know kfactory though), in the underlying kfactory, you can use [gf.kf.enclosure.extrude_path_dynamic](https://gdsfactory.github.io/kfactory/reference/kfactory/enclosure/#kfactory.enclosure.extrude_path_dynamic). This take a list of `gf.kdb.DPoint` as path and either a list of widths or a callable which returns a width profile base on the path length normalized to `0 -> 1`. Note: the main_layer must be a `gf.kdb.LayerInfo(layernumber, datatype)` object.

**Reply by smgitrep** [reply]
I am using the polygon approach in GDSFACTORY now, but will take a look at the KFACTORY function. Thanks for the help!
